test/Containers/test_macro.jl

```diff
@@ -279,4 +279,11 @@ function test_trailing_semicolon()
     return
 end
 
+function test_init_issue_4096()
+    x = Containers.DenseAxisArray(Any[1, 2, 3], Int[1, 2, 3])
+    @test sum(x) == 6
```

Comment:
Can you add a test with `@allocated ... == 0` to check that the dispatch is done a compile time ? It think it is still able to do this if-else at compile time